cluster_experiments/analysis_results.py

```diff
@@ -0,0 +1,109 @@
+from dataclasses import asdict, dataclass
+from typing import List
+
+import pandas as pd
+
+
+@dataclass
+class HypothesisTestResults:
+    """
+    A dataclass used to represent the results of a Hypothesis Test.
+
+    Attributes
+    ----------
+    metric_alias : str
+        The alias of the metric used in the test
+    control_variant_name : str
+        The name of the control variant
+    treatment_variant_name : str
+        The name of the treatment variant
+    control_variant_mean : float
+        The mean value of the control variant
+    treatment_variant_mean : float
+        The mean value of the treatment variant
+    analysis_type : str
+        The type of analysis performed
+    ate : float
+        The average treatment effect
+    ate_ci_lower : float
+        The lower bound of the confidence interval for the ATE
+    ate_ci_upper : float
+        The upper bound of the confidence interval for the ATE
+    p_value : float
+        The p-value of the test
+    std_error : float
+        The standard error of the test
+    dimension_name : str
+        The name of the dimension
+    dimension_value : str
+        The value of the dimension
+    alpha: float
+        The significance level of the test
+    """
+
+    metric_alias: str
+    control_variant_name: str
+    treatment_variant_name: str
+    control_variant_mean: float
+    treatment_variant_mean: float
+    analysis_type: str
+    ate: float
+    ate_ci_lower: float
+    ate_ci_upper: float
+    p_value: float
+    std_error: float
+    dimension_name: str
+    dimension_value: str
+    alpha: float
+
+
+class AnalysisPlanResults(pd.DataFrame):
```

Comment:
I don't love depending on a class we cannot control. do you think there's an alternative?

Reply:
I'd also consider making it a dataclass

Reply:
In case you want to make it a dataclass, what would the attribute look like? Because currently that purpose is served by the HypothesisTestResults class